flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/HeapKeyedStateBackend.java

```diff
@@ -622,20 +622,23 @@ public HeapSnapshotStrategy(
 
 			final Map<String, Integer> kVStateToId = new HashMap<>(stateTables.size());
 
-			final Map<StateTable<K, ?, ?>, StateTableSnapshot> cowStateStableSnapshots =
+			final Map<String, StateTableSnapshot> cowStateStableSnapshots =
 				new HashedMap(stateTables.size());
```

Comment:
Seems like I accidentally used `new HashedMap(...)` here instead of Java's `new HashMap<>(...)`. As you touch this spot already, maybe you can just also correct that?

Reply:
I will correct it in this PR.